sharedbuffers/mapped_struct.py

```diff
@@ -135,35 +135,35 @@
 @cython.cfunc
 @cython.inline
 def make_memoryview(a):
-    return memoryview(a).cast("B")
+    if python3:
+        return memoryview(a).cast("B")
```

Comment:
Can we avoid creating the memoryview?

``` python
if type(a) is not memoryview:
    a = memoryview(a)
```

There's probably a way to also avoid the cast... it should be a fast way, `a.format == "B"` works but it may be slow if it's python operations instead of native. I could not find a way to make it native.

I did a test, and it seems even without cython help it's quite faster to avoid the unnecessary operations:

``` python
import cython
  
_memoryview = cython.declare(object)
_memoryview = memoryview

def fbuffer(a):
    rv = a if type(a) is _memoryview else _memoryview(a)
    if rv.format != 'B':
        rv = rv.cast('B')
    return rv

def mbuffer(a):
    rv = a if type(a) is _memoryview else _memoryview(a)
    return rv.cast('B')

def sbuffer(a):
    return _memoryview(a).cast('B')
```

With a few timeits:

```
>>> import timeit, cython, pyximport
>>> pyximport.install(pyimport=True)
(<pyximport.pyximport.PyImporter object at 0x7ff42a518cc0>, <pyximport.pyximport.PyxImporter object at 0x7ff42a51ec18>)
>>> import ctest
>>> timeit.timeit(lambda:ctest.fbuffer(b"hola"), number=10000000) / 10.0
0.3661114629008807
>>> timeit.timeit(lambda:ctest.mbuffer(b"hola"), number=10000000) / 10.0
0.5427464027947281
>>> timeit.timeit(lambda:ctest.sbuffer(b"hola"), number=10000000) / 10.0
0.5551418196002487
>>> timeit.timeit(lambda:ctest.fbuffer(ctest.fbuffer(b"hola")), number=10000000) / 10.0
0.5739469211956021
>>> timeit.timeit(lambda:ctest.mbuffer(ctest.mbuffer(b"hola")), number=10000000) / 10.0
0.8258275890024379
>>> timeit.timeit(lambda:ctest.sbuffer(ctest.sbuffer(b"hola")), number=10000000) / 10.0
0.9486311056010891
```

It's clear that fbuffer wins by a healthy margin. It's a small difference (a fraction of a microsecond), but it's used a lot this make_memoryview thing, so I think it's worth paying attention to performance there.